src/main/java/pl/nogacz/checkers/board/Board.java

```diff
@@ -404,4 +425,109 @@ public static boolean isThisSameColor(Coordinates coordinates, PawnColor color)
     public static PawnClass getPawn(Coordinates coordinates) {
         return board.get(coordinates);
     }
+
+     class Menu extends JFrame implements ActionListener{
+        public static final int WIDTH = 500;
+        public static final int HEIGHT = 500;
+        private JButton Resume_button = new JButton("Resume");
+        private JButton NewGame_button = new JButton("New Game");
+        private JButton Exit_button = new JButton("EXIT");
+    
+        public Menu(Board board) {
+            //Initialize
+            super("Game Menu");
+            setSize(WIDTH,HEIGHT);
+            setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE); // dont need that but maybe someone want to remove undecorated
+            this.setUndecorated(true);
+            this.setBackground(new Color(0, 0, 0,100));
+            this.add(Resume_button);
+            this.add(NewGame_button);
+            this.add(Exit_button);
+            //Set Bounds
+            String exit_path = this.getClass().getClassLoader().getResource("button_exit.png").getFile();
+            String resume_path = this.getClass().getClassLoader().getResource("button_resume.png").getFile();
+            String new_game_path = this.getClass().getClassLoader().getResource("button_new-game.png").getFile();
+            Exit_button.setIcon(new ImageIcon(exit_path));
+            Resume_button.setIcon(new ImageIcon(resume_path));
+            NewGame_button.setIcon(new ImageIcon(new_game_path));
+            Resume_button.setBounds(150, 50, 225, 80);
+            NewGame_button.setBounds(150, 200, 280, 80);
+            Exit_button.setBounds(150, 350, 150, 80);
+            //Give them Func
+            Resume_button.addActionListener(this);
+            NewGame_button.addActionListener(this);
+            Exit_button.addActionListener(this);
+    
+            setLocationRelativeTo(null);
+            setLayout(null);
+            setVisible(true);
+    
+            this.addWindowListener(new WindowAdapter() {
+                public void windowClosing(WindowEvent e) {
+                    isMenuActive=false;
+                }
+            });
+        }
+    
+        @Override
+        public void actionPerformed(ActionEvent actionEvent) {
+            String command = actionEvent.getActionCommand();
+            System.out.println("hey");
+            if(command.equals("Resume")) {
+                System.out.println("pressed");
+                    isMenuActive=false;
+                    this.dispatchEvent(new WindowEvent(this,WindowEvent.WINDOW_CLOSING));
+            }else if(command.equals("New Game")) {
+                restart();
+            }else if(command.equals("EXIT")){
+                System.exit(0);
+            }
+        }
+    
+        public void restart()
+        {
+            Platform.runLater(() -> { 
+                for(Map.Entry<Coordinates, PawnClass> entry : board.entrySet()) {

```

Comment:
`for` içerde fakat geri kalanlar yanlış aling edilmiş. Alingmentı düzelt.

Reply:
Done